compiler/parser/src/parser/file.rs

```diff
@@ -66,13 +66,30 @@ impl ParserContext<'_> {
     ///
     pub fn parse_function_parameter_mode(&mut self) -> Result<ParamMode> {
         let public = self.eat(Token::Public);
```

Comment:
Would it be more optimal to chain conditional statements so that we are not making extra calls to `self.eat`
```rust
if self.eat(Token::Public).is_some() {
    return Ok(ParamMode::Public)
} else if self.eat(Token::Constant).is_some() { 
...
```

Reply:
Going this direction does prevent detection of multiple parameter tokens.
You can ignore this comment if chaining conditionals would allow syntax like
```rust
function foo(public constant bar: u8) {
...
```

Reply:
I can chain them together like so, but this prevents me from creating a more informative error message. This change would make it so the error message would just state `unexpected token constant` which isn't as informative IMO.

Reply:
Is it possible to give an informative error message without eating 3 tokens every time a function parameter is declared?

Reply:
Not really, since you need a span from the eaten ones to generate a helpful message. Also, the tokens are just representatives of what's already been parsed. It's just popping from a data structure and does not hinder performance at all.

Reply:
Also, it only eats the token if that is the token on top; even though it's 3 calls, it only pops one token if there's only one to match.

Reply:
Ah ok got it.